sw/device/tests/sim_dv/sysrst_ctrl_outputs_test.c

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_pinmux.h"
+#include "sw/device/lib/dif/dif_sysrst_ctrl.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+static dif_pinmux_t pinmux;
+static dif_sysrst_ctrl_t sysrst_ctrl;
+
+const uint32_t kTestPhaseTimeoutUsec = 100;
+
+enum {
+  kTestPhaseSetup = 0,
+  kTestPhaseLoopback = 1,
+  kTestPhaseOverrideSetup = 2,
+  kTestPhaseOverrideZeros = 3,
+  kTestPhaseOverrideOnes = 4,
+  kTestPhaseOverrideRelease = 5,
+  kTestPhaseOverrideAndLoopback = 6,
+  kTestPhaseDone = 7,
+};
+
+enum {
+  kAllZero = 0x0,
+  kAllOne = 0xff,
+  kLoopbackPartial = 0x5,
+  kNumMioInputs = 0x4,
+  kNumMioOutputs = 0x6,
+  kOutputNumPads = 0x8,
+};
+
+static const dif_pinmux_index_t kPeripheralInputs[] = {
+    kTopEarlgreyPinmuxPeripheralInSysrstCtrlAonKey0In,
+    kTopEarlgreyPinmuxPeripheralInSysrstCtrlAonKey1In,
+    kTopEarlgreyPinmuxPeripheralInSysrstCtrlAonKey2In,
+    kTopEarlgreyPinmuxPeripheralInSysrstCtrlAonPwrbIn,
+};
+
+static const dif_pinmux_index_t kInputPads[] = {
+    kTopEarlgreyPinmuxInselIob3,
+    kTopEarlgreyPinmuxInselIob6,
+    kTopEarlgreyPinmuxInselIob8,
+    kTopEarlgreyPinmuxInselIor13,
+};
+
+static const dif_pinmux_index_t kPeripheralOutputs[] = {
+    kTopEarlgreyPinmuxOutselSysrstCtrlAonKey0Out,
+    kTopEarlgreyPinmuxOutselSysrstCtrlAonKey1Out,
+    kTopEarlgreyPinmuxOutselSysrstCtrlAonKey2Out,
+    kTopEarlgreyPinmuxOutselSysrstCtrlAonPwrbOut,
+    kTopEarlgreyPinmuxOutselSysrstCtrlAonBatDisable,
+    kTopEarlgreyPinmuxOutselSysrstCtrlAonZ3Wakeup,
+};
+
+static const dif_pinmux_index_t kOutputPads[] = {
+    kTopEarlgreyPinmuxMioOutIob9, kTopEarlgreyPinmuxMioOutIor5,
+    kTopEarlgreyPinmuxMioOutIor6, kTopEarlgreyPinmuxMioOutIoc7,
+    kTopEarlgreyPinmuxMioOutIoc9, kTopEarlgreyPinmuxMioOutIob10,
+};
+
+static const dif_sysrst_ctrl_pin_t kSysrstCtrlOutputs[] = {
+    kDifSysrstCtrlPinKey0Out,           kDifSysrstCtrlPinKey1Out,
+    kDifSysrstCtrlPinKey2Out,           kDifSysrstCtrlPinPowerButtonOut,
+    kDifSysrstCtrlPinBatteryDisableOut, kDifSysrstCtrlPinZ3WakeupOut,
+    kDifSysrstCtrlPinEcResetInOut,      kDifSysrstCtrlPinFlashWriteProtectInOut,
+};
+
+// Test phase written by testbench.
+static volatile const uint8_t kTestPhase = 0;
+
+static void pinmux_setup(void) {
+  for (int i = 0; i < kNumMioInputs; ++i) {
+    CHECK_DIF_OK(
+        dif_pinmux_input_select(&pinmux, kPeripheralInputs[i], kInputPads[i]));
+  }
+  for (int i = 0; i < kNumMioOutputs; ++i) {
+    CHECK_DIF_OK(dif_pinmux_output_select(&pinmux, kOutputPads[i],
+                                          kPeripheralOutputs[i]));
+  }
+}
+
+static void wait_next_test_phase(void) {
```

Comment:
It may not be super obvious to a new reader that we are looking to see a `static const` value being modified. Perhaps provide some explanation (and path to the corresponding SV sequence that does the modification) in a comment. 

Also, a basic comment on what the other (not so obvious) static functions below do will be nice.

Reply:
Done